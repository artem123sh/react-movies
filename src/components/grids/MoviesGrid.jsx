import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { BACKGROUND } from '../../theme';
import GridControls from './MoviesGridControls';
import MoviesFoundLabel from './MoviesTotalFoundLabel';
import MovieGridItem from './MoviesGridItem';
import NoMovieFound from './NoMovieFound';

const StyledContainer = styled.div`
  flex-grow: 1;
  background: ${BACKGROUND};
  padding: 0 5%;
  display: flex;
  flex-direction: column;
`;

const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));;
  grid-row-gap: 3vh;
  grid-column-gap: 3vw;
  height: auto;
`;

const MoviesGrid = ({
  filter, sortBy, movies, handleEditMovie, toggleDeleteMovie, handleFilterChange,
  handleSortChange, totalAmount,
}) => (
  <StyledContainer>
    <GridControls
      filter={filter}
      sortBy={sortBy}
      handleFilterChange={handleFilterChange}
      handleSortChange={handleSortChange}
    />
    {movies.length ? (
      <>
        <MoviesFoundLabel number={totalAmount} />
        <Grid>
          {movies.map((movie) => (
            <MovieGridItem
              key={movie.id}
              id={movie.id}
              title={movie.title}
              releaseDate={movie.release_date}
              posterPath={movie.poster_path}
              genres={movie.genres}
              handleEditMovie={handleEditMovie}
              toggleDeleteMovie={toggleDeleteMovie}
            />
          ))}
        </Grid>
      </>
    ) : <NoMovieFound />}
  </StyledContainer>
);

MoviesGrid.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    release_date: PropTypes.string.isRequired,
    poster_path: PropTypes.string.isRequired,
    genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  })).isRequired,
  filter: PropTypes.string.isRequired,
  sortBy: PropTypes.string.isRequired,
  totalAmount: PropTypes.number.isRequired,
  handleEditMovie: PropTypes.func.isRequired,
  toggleDeleteMovie: PropTypes.func.isRequired,
  handleFilterChange: PropTypes.func.isRequired,
  handleSortChange: PropTypes.func.isRequired,
};

export default MoviesGrid;
