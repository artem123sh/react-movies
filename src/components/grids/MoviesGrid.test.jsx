import React from 'react';
import { render } from '@testing-library/react';
import MoviesGrid from './MoviesGrid';

describe('MoviesGrid', () => {
  test('renders MoviesGrid\'s snapshot', () => {
    const { asFragment } = render(<MoviesGrid
      filter=""
      sortBy="release_date"
      movies={[]}
      handleEditMovie={() => {}}
      toggleDeleteMovie={() => {}}
      handleFilterChange={() => {}}
      handleSortChange={() => {}}
      totalAmount={0}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
