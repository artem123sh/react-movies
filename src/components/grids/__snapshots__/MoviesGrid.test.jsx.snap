// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesGrid renders MoviesGrid's snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fzoNJl jTnteU"
  >
    <div
      class="sc-AxmLO jMiEjH"
    >
      <div>
        <button
          class="sc-Axmtr sc-fzplWN fyaKmV"
          type="button"
        >
          All
        </button>
        <button
          class="sc-Axmtr sc-fzplWN gKJFVO"
          type="button"
        >
          Documentary
        </button>
        <button
          class="sc-Axmtr sc-fzplWN gKJFVO"
          type="button"
        >
          Comedy
        </button>
        <button
          class="sc-Axmtr sc-fzplWN gKJFVO"
          type="button"
        >
          Horror
        </button>
        <button
          class="sc-Axmtr sc-fzplWN gKJFVO"
          type="button"
        >
          Crime
        </button>
      </div>
      <div
        class="sc-fzozJi wjHre"
      >
        <span
          class="sc-fzoLsD PYQVh"
        >
          Sort By
        </span>
        <div
          class="sc-AxiKw lkoKWO sc-fzpans cMhZxt"
        >
          <div>
            <button
              class="sc-AxjAm sc-AxhUy iFRYwR"
              type="button"
            >
              <span
                class="sc-AxgMl bkcfeM"
              >
                Release Date
              </span>
              <svg
                class="sc-AxheI ctYtXF"
                fill="#F65261"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-fzqNqU jIEkDE"
    >
      <span
        class="sc-fzqARJ sc-fzoyTs fWZkaA"
      >
        No Movies Found
      </span>
    </div>
  </div>
</DocumentFragment>
`;
